src/ASM/FiniteElement.C

```diff
@@ -80,15 +80,17 @@ bool MxFiniteElement::Jacobian (Matrix& Jac, const Matrix& Xnod,
                                 const std::vector<const BasisFunctionVals*>* bf,
                                 const std::vector<Matrix>* dNxdu)
 {
-  detJxW = utl::Jacobian(Jac,
-                         gBasis > 1 ? dMdX[gBasis-2] : dNdX, Xnod,
-                         bf ? (*bf)[gBasis-1]->dNdu : (*dNxdu)[gBasis-1]);
-  if (detJxW == 0.0) return false; // singular point
+  size_t nBasis = bf ? bf->size() : dNxdu->size();
+  if (gBasis <= nBasis) {
+    detJxW = utl::Jacobian(Jac,
+                           gBasis > 1 ? dMdX[gBasis-2] : dNdX, Xnod,
+                           bf ? (*bf)[gBasis-1]->dNdu : (*dNxdu)[gBasis-1]);
+    if (detJxW == 0.0) return false; // singular point
+  }
```

Comment:
Wouldn't it make more sense to just do `if (gBasis > nBasis) return false;` here? The calculations below will otherwise be meaningless since the `Jac` then is not set, or do you make the assumption it has the correct value as input (dangerous). Not quite sure what the intention is here.